verification/curator-service/ui/src/components/common-form-fields/Source.tsx

```diff
@@ -1,27 +1,151 @@
-import { FastField } from 'formik';
+import { Field, useFormikContext } from 'formik';
+
+import { Autocomplete } from '@material-ui/lab';
 import React from 'react';
 import { RequiredHelperText } from './FormikFields';
 import Scroll from 'react-scroll';
 import { TextField } from 'formik-material-ui';
+import { Typography } from '@material-ui/core';
+import axios from 'axios';
+import { throttle } from 'lodash';
 
 export default class Source extends React.Component<{}, {}> {
     render(): JSX.Element {
         return (
             <Scroll.Element name="source">
                 <fieldset>
                     <legend>Source</legend>
-                    <FastField
-                        label="Source URL"
-                        name="sourceUrl"
-                        type="text"
-                        required
-                        placeholder="https://..."
-                        component={TextField}
-                        fullWidth
-                    />
-                    <RequiredHelperText name={'sourceUrl'}></RequiredHelperText>
+                    <SourcesAutocomplete name="sourceUrl" required />
                 </fieldset>
             </Scroll.Element>
         );
     }
 }
+
+interface OriginData {
+    url: string;
+}
+
+interface SourceData {
+    _id: string;
+    origin: OriginData;
+}
+
+interface ListSourcesResponse {
+    sources: SourceData[];
+}
+
+interface SourcesAutocompleteProps {
+    name: string;
```

Comment:
optional: If these are always going to be set as the same thing for the SourcesAutocomplete we don't have to pass them in as props.

Reply:
I'd originally done this to allow us to avoid assumptions about the URL `Value` field name/whether this was required in the forms in which it appears, but you're right -- I'll scrap this now and can wire it back up if we need it.